### ISSUE TITLE
Unable to install on Hyper v1.0.0

### ISSUE THREAD

**Issue by devilleweppenaar** [dfrankland/hyper-sync-settings#19]
As suggested in #17, I tried enabling the `hyper-sync-settings` after upgrading to Hyper v1.0.0, but I get the following error on NPM install (also tried with Yarn):

```bash
$ npm i
npm WARN prefer global node-gyp@3.4.0 should be installed with -g

> nodegit@0.16.0 install /Users/deville.weppenaar/.hyper_plugins/node_modules/nodegit
> node lifecycleScripts/preinstall && node lifecycleScripts/install

[nodegit] Running pre-install script
[nodegit] Configuring libssh2.
{ Error: Command failed: /Users/deville.weppenaar/.hyper_plugins/node_modules/nodegit/vendor/libssh2/configure --with-libssl-prefix=/Users/deville.weppenaar/.hyper_plugins/node_modules/nodegit/vendor/openssl/openssl
/Users/deville.weppenaar/.hyper_plugins/node_modules/nodegit/vendor/libssh2/missing: Unknown `--is-lightweight' option
Try `/Users/deville.weppenaar/.hyper_plugins/node_modules/nodegit/vendor/libssh2/missing --help' for more information
configure: WARNING: 'missing' script is too old or missing
configure: error: No crypto library found!
Try --with-libssl-prefix=PATH
 or --with-libgcrypt-prefix=PATH
 or --with-wincng on Windows

    at ChildProcess.exithandler (child_process.js:206:12)
    at emitTwo (events.js:106:13)
    at ChildProcess.emit (events.js:191:7)
    at maybeClose (internal/child_process.js:877:16)
    at Socket.<anonymous> (internal/child_process.js:334:11)
    at emitOne (events.js:96:13)
    at Socket.emit (events.js:188:7)
    at Pipe._handle.close [as _onclose] (net.js:498:12)
  killed: false,
  code: 1,
  signal: null,
  cmd: '/Users/deville.weppenaar/.hyper_plugins/node_modules/nodegit/vendor/libssh2/configure --with-libssl-prefix=/Users/deville.weppenaar/.hyper_plugins/node_modules/nodegit/vendor/openssl/openssl' }
/Users/deville.weppenaar/.hyper_plugins/node_modules/nodegit/vendor/libssh2/missing: Unknown `--is-lightweight' option
Try `/Users/deville.weppenaar/.hyper_plugins/node_modules/nodegit/vendor/libssh2/missing --help' for more information
configure: WARNING: 'missing' script is too old or missing
configure: error: No crypto library found!
Try --with-libssl-prefix=PATH
 or --with-libgcrypt-prefix=PATH
 or --with-wincng on Windows

[nodegit] ERROR - Could not finish preinstall
{ Error: Command failed: /Users/deville.weppenaar/.hyper_plugins/node_modules/nodegit/vendor/libssh2/configure --with-libssl-prefix=/Users/deville.weppenaar/.hyper_plugins/node_modules/nodegit/vendor/openssl/openssl
/Users/deville.weppenaar/.hyper_plugins/node_modules/nodegit/vendor/libssh2/missing: Unknown `--is-lightweight' option
Try `/Users/deville.weppenaar/.hyper_plugins/node_modules/nodegit/vendor/libssh2/missing --help' for more information
configure: WARNING: 'missing' script is too old or missing
configure: error: No crypto library found!
Try --with-libssl-prefix=PATH
 or --with-libgcrypt-prefix=PATH
 or --with-wincng on Windows

    at ChildProcess.exithandler (child_process.js:206:12)
    at emitTwo (events.js:106:13)
    at ChildProcess.emit (events.js:191:7)
    at maybeClose (internal/child_process.js:877:16)
    at Socket.<anonymous> (internal/child_process.js:334:11)
    at emitOne (events.js:96:13)
    at Socket.emit (events.js:188:7)
    at Pipe._handle.close [as _onclose] (net.js:498:12)
  killed: false,
  code: 1,
  signal: null,
  cmd: '/Users/deville.weppenaar/.hyper_plugins/node_modules/nodegit/vendor/libssh2/configure --with-libssl-prefix=/Users/deville.weppenaar/.hyper_plugins/node_modules/nodegit/vendor/openssl/openssl' }
hyper-plugins@0.0.1 /Users/deville.weppenaar/.hyper_plugins
├── d@0.1.1  extraneous
├─┬ hyper-tab-icons@1.1.2 
│ ├── fuzzaldrin@2.1.0 
│ └─┬ react@15.4.1 
│   ├─┬ fbjs@0.8.6 
│   │ ├── core-js@1.2.7 
│   │ ├─┬ isomorphic-fetch@2.2.1 
│   │ │ └── whatwg-fetch@2.0.1 
│   │ ├─┬ promise@7.1.1 
│   │ │ └── asap@2.0.5 
│   │ └── ua-parser-js@0.7.12 
│   ├─┬ loose-envify@1.3.0 
│   │ └── js-tokens@2.0.0 
│   └── object-assign@4.1.0 
├── hyperborder@0.9.1 
├── hypercwd@0.1.6 
├─┬ hyperline@0.6.1 
│ ├─┬ color@0.11.4 
│ │ ├── clone@1.0.2 
│ │ ├─┬ color-convert@1.8.2 
│ │ │ └── color-name@1.1.1 
│ │ └── color-string@0.3.0 
│ ├── json-loader@0.5.4 
│ └── systeminformation@3.13.0 
├── hyperterm-atom-dark@1.1.2 
└─┬ hyperterm-lastpass@1.0.1 
  ├─┬ babel-polyfill@6.20.0 
  │ ├── core-js@2.4.1 
  │ └── regenerator-runtime@0.10.1 
  ├─┬ babel-runtime@6.20.0 
  │ └── core-js@2.4.1 
  ├─┬ bootstrap@4.0.0-alpha.5 
  │ ├── jquery@3.1.1 
  │ └── tether@1.4.0 
  ├─┬ lastpass@1.0.2 
  │ ├── bluebird@3.4.6 
  │ ├── es6-error@3.2.0 
  │ ├─┬ form-data@1.0.1 
  │ │ ├─┬ async@2.1.4 
  │ │ │ └── lodash@4.17.2 
  │ │ ├─┬ combined-stream@1.0.5 
  │ │ │ └── delayed-stream@1.0.0 
  │ │ └─┬ mime-types@2.1.13 
  │ │   └── mime-db@1.25.0 
  │ ├─┬ html-to-text@2.1.3 
  │ │ ├── he@1.1.0 
  │ │ ├── htmlparser@1.7.7 
  │ │ ├─┬ optimist@0.6.1 
  │ │ │ ├── minimist@0.0.8 
  │ │ │ └── wordwrap@0.0.3 
  │ │ ├── underscore@1.8.3 
  │ │ └─┬ underscore.string@3.3.4 
  │ │   ├── sprintf-js@1.0.3 
  │ │   └── util-deprecate@1.0.2 
  │ ├─┬ node-fetch@1.6.3 
  │ │ ├─┬ encoding@0.1.12 
  │ │ │ └── iconv-lite@0.4.15 
  │ │ └── is-stream@1.1.0 
  │ ├─┬ sanitize-filename@1.6.1 
  │ │ └─┬ truncate-utf8-bytes@1.0.2 
  │ │   └── utf8-byte-length@1.0.4 
  │ └─┬ xml2js@0.4.17 
  │   ├── sax@1.2.1 
  │   └── xmlbuilder@4.2.1 
  └── react-dom@15.4.1 

npm ERR! Darwin 16.1.0
npm ERR! argv "/Users/deville.weppenaar/.nvm/versions/node/v6.9.1/bin/node" "/Users/deville.weppenaar/.nvm/versions/node/v6.9.1/bin/npm" "i"
npm ERR! node v6.9.1
npm ERR! npm  v3.10.8
npm ERR! code ELIFECYCLE

npm ERR! nodegit@0.16.0 install: `node lifecycleScripts/preinstall && node lifecycleScripts/install`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the nodegit@0.16.0 install script 'node lifecycleScripts/preinstall && node lifecycleScripts/install'.
npm ERR! Make sure you have the latest version of node.js and npm installed.
npm ERR! If you do, this is most likely a problem with the nodegit package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     node lifecycleScripts/preinstall && node lifecycleScripts/install
npm ERR! You can get information on how to open an issue for this project with:
npm ERR!     npm bugs nodegit
npm ERR! Or if that isn't available, you can get their info via:
npm ERR!     npm owner ls nodegit
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     /Users/deville.weppenaar/.hyper_plugins/npm-debug.log
```

I verified that removing `hyper-sync-settings` from the package.json in `~/.hyper_plugins` leads to a successful npm install.

Please let me know if I can provide more info.

**Comment by dfrankland** [comment]
**You cannot manually install using `npm`**, unless you copy the `.npmrc` from this repo to `~/.hyper_plugins/`, be running Node.js v6 (if running Node.js v7, you must allow `nodegit` to build from source, using `libgcrypt`. Install it using `brew install libgcrypt`), and run `npm install` in `~/.hyper_plugins/`.

**The preferred method is to install with Hyper**, that way the proper environmental variables are loaded for `nodegit` to install. You may want to retry by removing `~/.hyper_plugins/node_modules` and using the `Update all plugins` menu option. It may take a while for all the dependencies to install, but everything should work after that.

**Comment by devilleweppenaar** [comment]
Thanks for the prompt response @dfrankland! Sorry, I neglected to mention that I first tried using the normal method of adding `hyper-sync-settings` to the `plugins` section of the hyper config file.

Doing this, I get the following error in the console:
```
bundle.js:669 Error: Cannot find module '/Users/deville.weppenaar/.hyper_plugins/node_modules/hyper-sync-settings'
    at Module._resolveFilename (module.js:455:15)
    at Function.Module._resolveFilename (/Applications/Hyper.app/Contents/Resources/electron.asar/common/reset-search-paths.js:35:12)
    at Function.Module._load (module.js:403:25)
    at require (internal/module.js:20:19)
    at paths.plugins.concat.map.err (file:///Applications/Hyper.app/Contents/Resources/app/dist/bundle.js:667:20)
    at Array.map (native)
    at loadModules (file:///Applications/Hyper.app/Contents/Resources/app/dist/bundle.js:660:54)
    at Object.<anonymous> (file:///Applications/Hyper.app/Contents/Resources/app/dist/bundle.js:765:1)
    at __webpack_require__ (file:///Applications/Hyper.app/Contents/Resources/app/dist/bundle.js:20:30)
    at Object.<anonymous> (file:///Applications/Hyper.app/Contents/Resources/app/dist/bundle.js:51874:73)
```

Then upon further investigation, I tried to manually run `npm install` from the plugins folder to get the error that is preventing the hyper-sync-settings from being installed.

**Comment by qndrey** [comment]
Faced with almost same issue.

```
Error: Cannot find module '../build/Debug/nodegit.node'
    at Module._resolveFilename (module.js:455:15)
    at Function.Module._resolveFilename
```

Any thoughts?

**Comment by matthewkturner** [comment]
Same issue as @andrewBalekha. 

```
bundle.js:669 Error: Cannot find module '../build/Debug/nodegit.node'
    at Module._resolveFilename (module.js:455:15)
    at Function.Module._resolveFilename (/opt/Hyper/resources/electron.asar/common/reset-search-paths.js:35:12)
    at Function.Module._load (module.js:403:25)
    at require (internal/module.js:20:19)
    at Object.<anonymous> (/home/matthew/.hyper_plugins/node_modules/nodegit/dist/nodegit.js:18:12)
    at Module._compile (module.js:556:32)
    at Object.Module._extensions..js (module.js:565:10)
    at Module.load (module.js:473:32)
    at tryModuleLoad (module.js:432:12)
    at Function.Module._load (module.js:424:3)
```

Installed via the preferred method mentioned above by @dfrankland on Ubuntu 16.04.

**Comment by dfrankland** [comment]
I think I found the problem with `nodegit`... Even if it is not installing from source, it still needs the dependencies to build from source. For the time being could you try to install one of the following packages?

For macOS:
```bash
brew install libgcrypt
```

For Linux:
```bash
apt install libssl-dev
```

Sorry, I think we need to do something different than just relying on `nodegit` 😢

**Comment by dfrankland** [comment]
Here's the issue on `nodegit` if anyone wants to know what's going on: https://github.com/nodegit/nodegit/issues/1166

**Comment by dfrankland** [comment]
This should be all fixed by `hyper-sync-settings@2.0.0` with PR https://github.com/dfrankland/hyper-sync-settings/pull/23
